### PR TITLE
[HttpKernel] Document the `terminate_on_cache_hit` option

### LINKED ISSUE #16919: [HttpCache] Do not call terminate() on cache hit
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46763
| PR author(s) | @Toflar
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #16919.

### CODE REVIEW

File: components/http_foundation.rst [hunk 1]
```diff
@@ -517,13 +517,18 @@ call::
         'immutable'        => true,
         'last_modified'    => new \DateTime(),
         'etag'             => 'abcdef',
+        'terminate_on_cache_hit' => false,
```
Comment (xabbuh) [inline components/http_foundation.rst]: We need to revert the changes in this document. The option is not part of the HTTP cache headers but only supported by our `HttpCache` class coming from the HttpKernel component.

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -144,6 +144,22 @@ which the cache can serve a stale response when an error is encountered
 (default: 60). This setting is overridden by the stale-if-error HTTP
 Cache-Control extension (see RFC 5861).
 
+terminate_on_cache_hit
```
Comment (xabbuh) [inline reference/configuration/framework.rst:147]: I think we made a mistake by placing this (and the other HTTP cache) options here. If I do not completely miss anything these options cannot be configured using the semantic configuration.

**Comment by javiereguiluz** [comment]
Thanks for your review Christian. I'm very confused, so I don't know how to move forward. Is this an internal thing that shouldn't be documented ... or is there something missing in the docs to make this option more useful? Thanks!

**Comment by javiereguiluz** [comment]
I'm closing this PR because I don't know how to move forward and it's better than others take over this. Thanks!

File: reference/configuration/framework.rst (hunk 2 above)
Comment (wouterj) [reply reference/configuration/framework.rst:147]: Since https://github.com/symfony/symfony/pull/37351 , we do allow the options to be configured through semantic configuration.

I've created https://github.com/symfony/symfony/pull/47630 to also add this new option to it.

**Comment by wouterj** [comment]
@javiereguiluz you can reopen this PR and only revert all changes from `components/http_foundation.rst`. All other changes are valid once https://github.com/symfony/symfony/pull/47630 is merged (and I'm certain that it will be merged before 6.2 stable).

**Comment by javiereguiluz** [comment]
I've reverted the wrong changes. Thanks for your review!

**Review by OskarStark — APPROVED** [review]
